SignalServiceKit/tests/Messages/Interactions/TSOutgoingMessageTest.m

```diff
@@ -17,51 +17,98 @@ @interface TSOutgoingMessageTest : SSKBaseTest
-                                                                attachmentIds:[NSMutableArray new]
-                                                             expiresInSeconds:10];
-    [message updateWithMessageState:TSOutgoingMessageStateSentToService];
-    XCTAssert(message.shouldStartExpireTimer);
+    TSOutgoingMessage *message =
+        [[TSOutgoingMessage alloc] initOutgoingMessageWithTimestamp:100
+                                                           inThread:self.thread
+                                                        messageBody:nil
+                                                      attachmentIds:[NSMutableArray new]
+                                                   expiresInSeconds:10
+                                                    expireStartedAt:0
+                                                     isVoiceMessage:NO
+                                                   groupMetaMessage:TSGroupMetaMessageUnspecified
+                                                      quotedMessage:nil
+                                                       contactShare:nil];
+    [[OWSPrimaryStorage sharedManager].dbReadWriteConnection
+        readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+            [message updateWithSentRecipient:self.contactId transaction:transaction];
+            XCTAssertTrue([message shouldStartExpireTimerWithTransaction:transaction]);
+        }];
 }
 
 - (void)testShouldNotStartExpireTimerWithUnsentMessage
 {
-    TSOutgoingMessage *message = [[TSOutgoingMessage alloc] initWithTimestamp:100
-                                                                     inThread:self.thread
-                                                                  messageBody:nil
-                                                                attachmentIds:[NSMutableArray new]
-                                                             expiresInSeconds:10];
-    [message updateWithMessageState:TSOutgoingMessageStateUnsent];
-    XCTAssertFalse(message.shouldStartExpireTimer);
+    TSOutgoingMessage *message =
+        [[TSOutgoingMessage alloc] initOutgoingMessageWithTimestamp:100
+                                                           inThread:self.thread
+                                                        messageBody:nil
+                                                      attachmentIds:[NSMutableArray new]
+                                                   expiresInSeconds:10
+                                                    expireStartedAt:0
+                                                     isVoiceMessage:NO
+                                                   groupMetaMessage:TSGroupMetaMessageUnspecified
+                                                      quotedMessage:nil
+                                                       contactShare:nil];
+    [[OWSPrimaryStorage sharedManager].dbReadWriteConnection
```

Comment:
Should we put a `self.readWriteWithBlock` on BaseSignalTest (TIOLI)

Reply:
Done. This DRY'd a lot of code.